### ISSUE TITLE
Validate specsim sky brightness against DES photometry

### ISSUE THREAD

**Issue by moustakas** [desihub/specsim#55]
This is a parallel issue to the SDSS-centric #40 (which was originally spun off from #9) focused on using DES photometry (using a file originally provided by @annis) to validate whether the 
lunar/sky model currently implemented in `specsim` accurately captures measured variations in sky brightness, especially during non-photometric conditions with the moon up.

The datafile can be grabbed [here](https://gist.github.com/moustakas/aad634541a3940faa2c9b8be5ecd1170).  The file should be self-documenting except note that the column `trans` is in magnitudes.  Also there may be more data from subsequent years of DES observations but I haven't corresponded with Jim about this since late-2015.

Let's try to post updates and progress on this topic here.

**Comment by belaa** [comment]
Here is a plot of sky brightness as predicted by `specsim` vs. measured sky brightness for a subset of DECaLS BGS data found [here](https://desi.lbl.gov/trac/wiki/MilkyWayWG/SkyBrightness). 

What is shown includes the following cuts: 

1)  Moon altitude > 10
2) Sun altitude < -15
3) 0.75 < Transparency  < 1.3

The model does a good job of predicting sky+moon in the g- and r-bands, although there is a slight offset in the g-band where `specsim` seems to under-estimate the sky brightness. I’m looking into whether this could be due to how `specsim` models moon reflectivity at lower wavelengths. It seems reasonable for there to be some scatter in the z-band due to the presence of OH lines, but I’m not sure exactly how much scatter to expect. 

![decals_sky_brightness](https://cloud.githubusercontent.com/assets/9096399/26137448/0fda3a78-3a77-11e7-9f84-88b8bacb46ce.png)

**Comment by dkirkby** [comment]
@moustakas Should Bela post this update to the BGS mailing list or what's the best way to follow up and get feedback?

**Comment by moustakas** [comment]
Yes, thank you for the ping @dkirkby!  @belaa if you could post this update to BGS and also possibly MWS there should be great interest.  (One minor request is to make all the axes consistent across the subplots, including the colorbar.)

**Comment by belaa** [comment]
@moustakas Here is a plot with the axes adjusted.

![decals_sky_brightness2](https://cloud.githubusercontent.com/assets/9096399/26264155/aa0b231c-3c90-11e7-90a9-aa440f3f30ae.png)

**Comment by ShaunMCole** [comment]
Belaa, this is really great to see.  I'd like to see histograms of the residuals, measured-model, and perhaps how they vary with transparency cut. Is the offset in the g band understood? -- Shaun

**Comment by belaa** [comment]
@ShaunMCole Here are the residuals, which contain all the original cuts (including transparency):
 
![residuals](https://cloud.githubusercontent.com/assets/9096399/26320313/4c8d34f2-3ed7-11e7-832b-2a2f796c4edb.png)

And here is a plot that shows how the data varies with transparency (this plot includes cuts on sun and moon altitude, but not transparency): 

![transparency](https://cloud.githubusercontent.com/assets/9096399/26320433/b701a642-3ed7-11e7-910b-8001b47a32a0.png)

Do you by any chance know the units for transparency? I gathered it to be the percentage of expected light that reaches us, but I may be wrong.

The residuals show an offset in both the g- and z-bands. Right now, `specsim` assumes a uniform moon reflectivity for all wavelengths. I wonder if the offset could be related to this assumption.

**Comment by moustakas** [comment]
According to my notes from @annis the units of `trans` are magnitudes (despite what's written in the header of the ASCII file).

**Comment by belaa** [comment]
Thanks @moustakas. I see you mentioned that in your note earlier. Some of the values for transparency were 99.0 and above so I was a little confused. Were these observations for which the transparency couldn't be determined?

I've adjusted the color bars in each band to correspond to values within a percentile range that excludes transparency >=99.0. So points in the plot that are above/below this range are set to the max/min color. 

![transparency](https://cloud.githubusercontent.com/assets/9096399/26325514/0bfe4c36-3eec-11e7-90e2-8f7a5b8a80b8.png)

**Comment by janewman-pitt-edu** [comment]
I think if you plot exposure time (if it's available) vs. transparency that'd help you see if it's bad or good.

I think to first order it looks like g has a simple offset, r is mostly OK, but z has large deviations that just aren't captured by the model (i.e. the model matches the best-case scenario mostly well with a modest offset, but the worst case is a whole lot worse).  I'm a little surprised the excursions only get to a magnitude though, I've observed on a cloudy moony night and believe it was getting brighter than that at bad times vs good.

**Comment by dkirkby** [comment]
UPDATE: the specsim-predicted surface brightnesses in the plots above are under-estimated because they used a filter curve that includes extinction through an airmass 1.2 atmosphere, but extinction has already been applied to the spectrum being convolved with the filter.  This is at least part of the reason for the systematic offset in the g-band comparison.  As a rough estimate, this should bring the predictions down by 0.25 (g) 0.12 (r) 0.13 (z).
